packages/pigeon/lib/dart_generator.dart

```diff
@@ -98,11 +107,11 @@ void _writeCodec(Indent indent, String codecName, Api api, Root root) {
           indent.write('case ${customClass.enumeration}: ');
           indent.writeScoped('', '', () {
             indent.writeln(
-                'return ${customClass.name}.decode(readValue(buffer)!);');
+                'return ${customClass.name}.decode(readValue(buffer)! as List<Object?>);');
```

Comment:
Why the cast here? `decode` still takes an `Object` (I assume it's deliberately vague since it's a public method, so using just `Object` lets us make changes like this without breaking the signature.)

Reply:
Just old cruft.